contrib/testgen/gen_key_io_test_vectors.py

```diff
@@ -114,8 +109,10 @@ def is_valid(v):
     '''Check vector v for validity'''
     if len(set(v) - set(b58chars)) > 0:
         return is_valid_bech32(v)
-    result = b58decode_chk(v)
-    if result is None:
+    try:
+        payload, version = base58_to_byte(v)
+        result = bytes([version]) + payload
+    except AssertionError:  # thrown if checksum doesn't match
```

Comment:
Catching an AssertionError for expected error handling is a bit strange. Normally one'd use `ValueError` or such for invalid values.

Reply:
Agreed that AssertionError is too generic for this case, I added a commit which changes this to `ValueError`, as suggested.